src/plugins/auth/auth.ts

```diff
@@ -262,6 +289,39 @@ const plugin: FastifyPluginAsync<AuthPluginOptions> = async (fastify, options) =
       },
     );
 
+    // login with password
+    fastify.post<{ Body: { email: string; password: string } }>(
+      '/login-password',
+      { schema: passwordLogin },
+      async ({ body, log }, reply) => {
+        const email = body.email.toLowerCase();
+        const task = memberTaskManager.createGetByTask(GRAASP_ACTOR, { email });
+        const [member] = await runner.runSingle(task, log);
+
+        if (!member) {
+          const { email } = body;
+          log.warn(`Login attempt with non-existent email '${email}'`);
+          return reply.status(StatusCodes.NOT_FOUND).send(ReasonPhrases.NOT_FOUND);
+        }
+        if (member.password === null) {
+          log.warn('Login attempt with non-existent password');
+          return reply.status(StatusCodes.NOT_ACCEPTABLE).send(ReasonPhrases.NOT_ACCEPTABLE);
+        }
+        const verified = await verifyCredentials(member, body);
+        if (!verified) {
+          reply.status(StatusCodes.UNAUTHORIZED).send(ReasonPhrases.UNAUTHORIZED);
+        } else {
```

Comment:
So we don't factor out?

Reply:
Thought this is what you meant in previous comments. Here we try to avoid encapsulating ifs and use the `return reply`
First check error cases (check member and password exists, as it is done in other functions above) and `return`.
Then verify credential and check success cases. 
We can try to refactor again if you consider it necessary.